app/com/gu/floodgate/DynamoDBTable.scala

```diff
@@ -62,6 +62,32 @@ trait DynamoDBTable[T] extends StrictLogging {
     promise.future
   }
 
+  def getItems(id: String, keyName: String) = {
+    val promise = Promise[List[T]]()
+
+    val request = new QueryRequest().withTableName(tableName)
+      .withKeyConditionExpression(s"$keyName = :$keyName")
+      .withExpressionAttributeValues(
+        Map(s":$keyName" -> new AttributeValue().withS(id)))
+
+    val responseHandler = new AsyncHandler[QueryRequest, QueryResult] {
+
+      override def onError(e: Exception) = {
+        logger.warn(s"Could not retrieve item with hash key: $id: ${e.getMessage} ")
+        promise.failure(e)
+      }
+
+      override def onSuccess(request: QueryRequest, result: QueryResult) = {
+        val item: List[T] = result.getItems.asScala.toList.map(f => fromItem(f.toMap))
+        promise.success(item)
+      }
+
+    }
+
+    dynamoDB.queryAsync(request, responseHandler)
+    promise.future
```

Comment:
I think this "create a promise, create an async handler, make a request, return the future" process can be abstracted. Something like:

```scala
def makeAsyncDynamoRequest[Req, Resp, ReturnType](request: Req, makeReq: (req: Req, handler: AsyncHandler[Req, Resp]) => Unit, convertResult: Resp => ReturnType, errorMsg: Exception => String): Future[ReturnType] = {
  val promise = Promise[ReturnType]()
  val handler = new AsyncHandler[Req, Resp] {
    override def onError(e: Exception) = {
      logger.warn(errorMsg)
      promise.failure(e)
    }
    override def onSuccess(request: Req, result: Resp) = 
      promise.success(convertResult(result))
  }
  makeReq(request, handler)
  promise.future
}

val request = new QueryRequest().withTableName(tableName)
    .withKeyConditionExpression(s"$keyName = :$keyName")
    .withExpressionAttributeValues(Map(s":$keyName" -> new AttributeValue().withS(id)))
val makeReq = (req: QueryRequest, handler: AsyncHandler[QueryRequest, QueryResult]) => dynamoDB.queryAsync(req, handler)
val convertResult = (result: QueryResult) => result.getItems.asScala.toList.map(f => fromItem(f.toMap))
val errorMsg = (e: Exception) => s"Could not retrieve item with hash key: $id: ${e.getMessage} "
makeAsyncDynamoRequest(request, makeReq, convertResult, errorMsg)
```

(Untested, probably doesn't compile!)